### PR TITLE
Copy `cortex/pkg/distributor` requirements into Tempo

### PR DESCRIPTION
**Pull request by zalegrala**
**What this PR does**:

As part of the migration away from `cortex`, here we copy what we need
to satisfy the distributor.

### CODE REVIEW

**Comment by zalegrala** [comment]
That cleans it up quite a bit. :)

File: modules/distributor/config.go [hunk 1]
```diff
@@ -4,10 +4,10 @@ import (
 	"flag"
 	"time"
 
-	cortex_distributor "github.com/cortexproject/cortex/pkg/distributor"
 	"github.com/grafana/dskit/flagext"
 	"github.com/grafana/dskit/ring"
 	ring_client "github.com/grafana/dskit/ring/client"
+	tempo_distributor "github.com/grafana/tempo/pkg/distributor"
```
Comment (joe-elliott) [inline modules/distributor/config.go:10]: no reason to give this an alias. just leave it `distributor`

**Review by joe-elliott — COMMENTED** [review]
one nit. looks good.

File: modules/distributor/config.go (hunk 1 above)
Comment (zalegrala) [reply modules/distributor/config.go:10]: I just didn't like shadowing the package name.

Comment (joe-elliott) [reply modules/distributor/config.go]: I didn't realize that you had copied these to `/pkg`. See comment here: https://github.com/grafana/tempo/pull/1251#discussion_r792660048

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**